### ISSUE TITLE
Column value inaccessible for Transformer Expressions during import/export

### ISSUE THREAD

**Issue by RivenSkaye** [dbeaver/dbeaver#20335]
### Description

I'm trying to migrate a database from MS Access to MariaDB and I'm facing a problem getting the column's value in the Expression field in the import/export builder. MS Access has some... interesting variation when it comes to data types and this database has been used for more time than anyone with IT experience has been employed. This leads to some of these fields internally being stored in rather interesting ways that MS Access somehow understands are booleans, even in forms.
- BOOLEAN
- VARCHAR(1) which has a localized y/n value (e.g. j/n for Dutch)
- VARCHAR(N) where N is the length of the longest of the two options (3 for yes/no, 3 for ja/nee)
- TINYINT being either nonzero for true, or 0 for false
- VARCHAR(1) being either `"0"` or `"1"` (yes, numeric strings)

There are also column names with question marks and spaces, to make things worse.

Attempting to migrate the database, this causes a problem converting the textual representations. So I figured I'd attempt using the JEXL Expression option in the data export wizard. The problem here being that I can't seem to get the values currently populating the cells in the database,

I've tried several variations of `not ( 'column name?' =~ "[nN0].*")` and `not (${value} =~ "[nN0].*")` after reading through the docs and seeing some examples for e.g. presenting URL views. I've also tried prefixing the quoted identifier with a `.` to no avail.

The only think providing a _different_ error message than the constant parser errors for unexpected characters and identifiers was `not (['column name?'] =~ "[nN0].*")`, producing
![image](https://github.com/dbeaver/dbeaver/assets/14993038/7dc657c5-abfd-444c-9114-a966787f8f9c)

### DBeaver Version

Community Edition 23.1.0

### Operating System

Windows 10

### Database and driver

Using UCanAccess with Jackcess-encrypt
2003 format MS Access MWB

### Steps to reproduce

1. Create MS Access database
2. Create table with spaces and/or question marks in column names with Access's own yes/no type
3. Try to export the table to a proper DBMS

### Additional context

Everything except for the MariaDB server are running on the same machine. Nothing changed about the DBeaver install except for changing a single configuration entry `-Xmx6144m` (running 500 rows on 2 threads already runs mem consumption up to 5.5GB, there are older issues about DBeaver OOM'ing)

**Comment by RivenSkaye** [comment]
Some additional, probably useful info is that I _can_ grab the value of other columns. E.g. the ID column and others without spaces or anything. It seems to purely be poor column naming that seems to be making it impossible to grab these values

**Comment by RivenSkaye** [comment]
Okay, final update on this.
in desparation before dropping the migration altogether, I decided to try `get('column name?')` in order to acquire usable values from this and _that_ actually worked. Not sure if it's intended to work or not (the [JEXL reference](https://commons.apache.org/proper/commons-jexl/reference/syntax.html) doesn't list it) but this is something that might be useful to add to any documentation that gets written on the Transformation expression functionality. I verified both the data (successfully got errors about parsing `"n"` as an int, as expected) and the ability to subsequently process it (`get('column name?') !~ "[n0N].*" ? 1 : 0` returns correct values) leaving only two grievances here, both of which I think should be considered a bug:
1. If you don't use a ternary operator, it goes back to erroring about not supporting booleans. This should be rather easy to support as all database systems have standards for it. E.g. MySQL/MariaDB provide the `BOOL` alias and the `TRUE` and `FALSE` constants that resolve to `TINYINT` values. Other options include getting the Boolean's value field or making a static check depending on the target database's format. In most if not all cases, casting to int would probably suffice.
2. There is no way to access these column names unless using `get` on what is effectively just the entire context scope. Some form of (documented) standardized name mangling, however crude, would go a long way here. The easiest I could think of would be to match column names to the JEXL provided legal identifier patterns and blindly replacing all offending characters with either an underscore or dollar sign or just removing the offending characters.

**Comment by ShadelessFox** [comment]
Hello @RivenSkaye,

Thanks for the report.

In your first example, you are trying to access row values using string literals (`'column name?'`), which is incorrect as it denotes a string, not a reference to a row value.

As for the booleans, it looks like a bug.

**Comment by RivenSkaye** [comment]
Yeah, I get that it's a string literal, it was one of many desperate attempts at getting at the variables. Other things I tried included substitution and omitting of illegal chars in the identifiers, `.['ident']` and similar in the hopes of convincing JEXL to grab something that _should_ be in scope.

Should I open a different issue for either boolean support or the variable naming problem? Or is the latter a case of `wontfix`?

**Comment by ShadelessFox** [comment]
Please keep the issue open. This issue is a result of missing documentation from our side.

**Comment by BloodyRain2k** [comment]
I feel like this can need a little bump, because I ran into the same issue. I got an SQLite Table where some boolean columns are string booleans, and I have no clue how I am even meant to write the JEXL expression to convert one into the other, because I can't find a single example for anything : (

Edit: ok, I finally figured it out, but holy hell... The fact that I had to do half a dozen clicks to edit the statement each attempt to get the prompt open again was seriously grinding my gears at the end...
Anyways, the actual expression was extremely simple after I finally found it out: `{column_to_check} == "true" ? 1 : 0`
In my case `has_comments == "true" ? 1 : 0`, that's already enough to convert the booleans. But for unknown reasons did `if (has_comments == "true") { return 1 } else { return 0 }` not work...

**Comment by battmush** [comment]
It's worth noting that this is still (24.2.5.202411171748) an issue, and unfortunately the JavaBean get() property workaround does not work for columns which contain a pound sign ('#').

Is there really no way in JEXL to implicitly refer to the column name variable you're stating the transform for?

**Comment by RivenSkaye** [comment]
> Edit: ok, I finally figured it out, but holy hell... The fact that I had to do half a dozen clicks to edit the statement each attempt to get the prompt open again was seriously grinding my gears at the end... Anyways, the actual expression was extremely simple after I finally found it out: `{column_to_check} == "true" ? 1 : 0` In my case `has_comments == "true" ? 1 : 0`, that's already enough to convert the booleans. But for unknown reasons did `if (has_comments == "true") { return 1 } else { return 0 }` not work...

Ah, sorry about missing this reply. It should also be pretty easy to modify the oneliner I'd initially provided as a working solution: `get('column name?') !~ "[n0NfF].*" ? 1 : 0` (add in lower- and uppercase F for matching) which should catch a broader range of cases where a non-valid input would end up getting interpreted as a truthy value in a lot of cases, whereas your solution will consider any other non-empty string to be falsy. Should keep the migrated entries more consistent in application behavior for most cases (unless the code backed by your SQLite DB does the same string comparison)
